examples/socketio/src/application.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-socketio
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BootMixin} from '@loopback/boot';
+import {ApplicationConfig} from '@loopback/core';
+import {RepositoryMixin} from '@loopback/repository';
+import {ServiceMixin} from '@loopback/service-proxy';
+import {SocketIoApplication, SocketIoServer} from '@loopback/socketio';
+import debugFactory from 'debug';
+import {SocketIoController} from './controllers';
+
+const debug = debugFactory('loopback:example:socketio:demo');
+
+export {ApplicationConfig};
+
+export class SocketIoExampleApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(SocketIoApplication)),
+) {
+  readonly ioServer: SocketIoServer;
+
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    this.projectRoot = __dirname;
+
+    this.socketServer.use((socket, next) => {
+      debug('Global middleware - socket:', socket.id);
+      console.log('global');
+      next();
+    });
+
+    const ns = this.socketServer.route(SocketIoController);
+    ns.use((socket, next) => {
+      debug(
+        'Middleware for namespace %s - socket: %s',
+        socket.nsp.name,
+        socket.id,
+      );
+      next();
+    });
+  }
```

Comment:
We should define middleware chain for socketio too. Similar as REST or gRPC.

Reply:
Do you mind like `MiddlewareSequence` in RestCrud example? Please can you tell me where is the sample of gRCP implementation?

Reply:
oohhh god, `MiddlewareSequence` isn't in rest-crud example, it is in `rest` package, that is what i wanted to say. sorry.